bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -102,15 +113,48 @@ private boolean isResultExpired(DiscoveryResult result, long now) {
     private Set<InboxListener> listeners = new CopyOnWriteArraySet<>();
 
     private DiscoveryServiceRegistry discoveryServiceRegistry;
+
     private ThingRegistry thingRegistry;
+
     private ManagedThingProvider managedThingProvider;
 
+    private ThingTypeRegistry thingTypeRegistry;
+
+    private ConfigDescriptionRegistry configDescRegistry;
+
     private Storage<DiscoveryResult> discoveryResultStorage;
 
     private ScheduledFuture<?> timeToLiveChecker;
 
     private EventPublisher eventPublisher;
 
+    private List<ThingHandlerFactory> thingHandlerFactories = new CopyOnWriteArrayList<>();
+
+    @Override
+    public Thing approve(ThingUID thingUID, String label) {
+        if (thingUID == null) {
+            throw new IllegalArgumentException("Thing UID must not be null");
+        }
+        List<DiscoveryResult> results = get(new InboxFilterCriteria(thingUID, null));
+        if (results.isEmpty()) {
+            throw new IllegalArgumentException("No Thing with UID "+ thingUID.getAsString() +" in inbox");
```

Comment:
did you apply the formatter? There seems to be spaces missing around the + signs.